### PR TITLE
Add getparam to service, hostgroup, servicegroup, contact and contactgroup

### PR DESCRIPTION
**Pull request by oriolclosa**
Function getparam has been added to service, hostgroup, servicegroup, contact and contactgroup objects. The function on the host object has also been modified to include other parameters left on the first implementation.

I have checked the function in all four objects with all the parameters in a random host and this was the result.

### Host object
```php
acknowledgement_timeout => acknowledgement_timeout : 
action_url => action_url : 
activate => activate : 1
active_checks_enabled => active_checks_enabled : 2
alias => alias : 
cg_additive_inheritance => cg_additive_inheritance : 0
check_command => check_command : 
check_command_arguments => Please check that your parameters are valid
check_freshness => check_freshness : 2
check_period => check_period : 
comment => comment : 
contact_additive_inheritance => contact_additive_inheritance : 0
description => description : TEST3
event_handler => event_handler : 
event_handler_arguments => event_handler_arguments : 
event_handler_enabled => event_handler_enabled : 2
first_notification_delay => first_notification_delay : 
flap_detection_enabled => flap_detection_enabled : 2
freshness_threshold => freshness_threshold : 
high_flap_threshold => high_flap_threshold : 
icon_image => icon_image : 
icon_image_alt => icon_image_alt : 
id => id : 300
inherit_contacts_from_host => inherit_contacts_from_host : 1
is_volatile => is_volatile : 2
locked => locked : 0
low_flap_threshold => low_flap_threshold : 
max_check_attempts => max_check_attempts : 
normal_check_interval => normal_check_interval : 
notes => notes : 
notes_url => notes_url : 
notifications_enabled => notifications_enabled : 2
notification_interval => notification_interval : 
notification_options => notification_options : 
notification_period => notification_period : 
notification_period => notification_period : 
obsess_over_service => obsess_over_service : 2
parallelize_check => parallelize_check : 2
passive_checks_enabled => passive_checks_enabled : 2
process_perf_data => process_perf_data : 2
recovery_notification_delay => recovery_notification_delay : 
register => register : 1
retain_nonstatus_information => retain_nonstatus_information : 2
retain_status_information => retain_status_information : 2
retry_check_interval => retry_check_interval : 
stalking_options => stalking_options : 
template => Please check that your parameters are valid
use_only_contacts_from_host => use_only_contacts_from_host : 
```

### Service object
```php
2d_coords => 2d_coords : 
3d_coords => 3d_coords : 
acknowledgement_timeout => acknowledgement_timeout : 
action_url => action_url : 
activate => activate : 1
activate => activate : 1
active_checks_enabled => active_checks_enabled : 2
address => address : 127.0.0.1
alias => alias : TEST1
cg_additive_inheritance => cg_additive_inheritance : 0
checks_enabled => checks_enabled : 2
check_command => check_command : 
check_command_arguments => check_command_arguments : 
check_freshness => check_freshness : 2
check_interval => check_interval : 
check_period => check_period : 
comment => comment : 
contact_additive_inheritance => contact_additive_inheritance : 0
event_handler => event_handler : 
event_handler_arguments => event_handler_arguments : 
event_handler_enabled => event_handler_enabled : 2
event_handler_enabled => event_handler_enabled : 2
first_notification_delay => first_notification_delay : 
flap_detection_enabled => flap_detection_enabled : 2
flap_detection_options => flap_detection_options : 
freshness_threshold => freshness_threshold : 
geo_coords => geo_coords : 
high_flap_threshold => high_flap_threshold : 
high_flap_threshold => high_flap_threshold : 
icon_image => icon_image : 
icon_image_alt => icon_image_alt : 
id => id : 425
location => location : 
locked => locked : 0
low_flap_threshold => low_flap_threshold : 
low_flap_threshold => low_flap_threshold : 
max_check_attempts => max_check_attempts : 
name => name : TEST1
notes => notes : asdasdas
notes_url => notes_url : 
notifications_enabled => notifications_enabled : 2
notification_interval => notification_interval : 
notification_options => notification_options : 
notification_period => notification_period : 
obsess_over_host => obsess_over_host : 2
passive_checks_enabled => passive_checks_enabled : 2
process_perf_data => process_perf_data : 2
recovery_notification_delay => recovery_notification_delay : 
register => register : 1
retain_nonstatus_information => retain_nonstatus_information : 2
retain_status_information => retain_status_information : 2
retry_check_interval => retry_check_interval : 
snmp_community => snmp_community : 
snmp_version => snmp_version : 
stalking_options => stalking_options : 
statusmap_image => statusmap_image : 
timezone => timezone : 
```

### Hostgroup object
```php
action_url => action_url : 
activate => activate : 1
alias => alias : TEST1
comment => comment : 
geo_coords => geo_coords : 
icon_image => icon_image : 
icon_image_alt => icon_image_alt : 
id => id : 425
name => name : TEST1
notes => notes : asdasdas
notes_url => notes_url : 
statusmap_image => statusmap_image : 
```

### Servicegroup object
```php
activate => activate : 1
alias => alias : TEST1
comment => comment : 
geo_coords => geo_coords : 
id => id : 425
name => name : TEST1
```

### Contact object
```php
activate => activate : 1
address1 => address1 : 
address2 => address2 : 
address3 => address3 : 
address4 => address4 : 
address5 => address5 : 
address6 => address6 : 
admin => admin : 1
alias => alias : admin
ar_id => ar_id : 
authtype => authtype : 
charset => charset : 
comment => comment : 
default_page => default_page : 
email => email : admin@admin.com
enable_notifications => enable_notifications : 1
hostnotifopt => hostnotifopt : 
hostnotifperiod => hostnotifperiod : 24x7
id => id : 1
language => language : 
ldap_dn => ldap_dn : 
name => name : admin_admin
pager => pager : 
reach_api => reach_api : 0
reach_api_rt => reach_api_rt : 0
register => register : 1
servicenotifopt => servicenotifopt : 
svcnotifperiod => svcnotifperiod : 24x7
template => template : 
type_msg => type_msg :
```

### Contactgroup object
```php
activate => activate : 1
alias => alias : Guests Group
ar_id => ar_id : 
comment => comment : 
id => id : 3
ldap_dn => ldap_dn : 
name => name : Guest
type => type : local
```

As shown, all parameters can be retrieved. I also tried to follow the coding style guide. Please be aware that this is my first PR to the Centreon repository and although I have some knowledge of the platform I had never looked it at this level.

### CODE REVIEW

File: www/class/centreon-clapi/centreonHost.class.php [hunk 1]
```diff
@@ -358,28 +358,39 @@ public function getparam($parameters = null)
...
             'event_handler_enabled',
+            'event_handler_enabled',
             'first_notification_delay',
             'flap_detection_enabled',
             'flap_detection_options',
-            'host_high_flap_threshold',
-            'host_low_flap_threshold',
+            'freshness_threshold',
+            'high_flap_threshold',
+            'high_flap_threshold',
```
Comment (sc979) [inline www/class/centreon-clapi/centreonHost.class.php]: isn't it `low_flap_threshold`

Comment (oriolclosa) [reply www/class/centreon-clapi/centreonHost.class.php]: Yes, you're right, I duplicated both.

**Comment by smutel** [comment]
Hello,

Before merging this PR, please take a look at this one: #7557.
I think I resolve a bug on the getparam of object host and the code of this PR is based on what was done previously for object host.

Thanks.

**Comment by lpinsivy** [comment]
Branch 2.8.x is no more maintained.
Severals getparams have been included.

**Pull request closed without merging**